### ISSUE TITLE
Default Mod Compatability for Weapon-like and Armor-like tools?

### ISSUE THREAD

**Issue by Dragnier** [TheXFactor117/Levels#77]
Is there a way for you to auto-populate certain tools into the level system if the correct mod is found installed? For example, Mekanism has items that could potentially level but do not; like the armored jetpack and freerunners. Actually, even the Mekanism extra armor types seem not to level. Another couple of mods I've seen that have normal-ish weapons/armors which seem not to level are Cyberware and Weapon Case Loot.

**Comment by TheXFactor117** [comment]
Yes, I have been looking into ways to get that to work. Although it might come to me to manually implement each mod, in which case would take a lot of work that I don't necessarily want to do.

I'll look into ways to implement it though.

**Comment by Dragnier** [comment]
I seem to remember there used to be something similar to the following, but I thought I'd mention it since I can't find it anymore.

If you were to create a config option allowing users to add weapons from mods as types, like the Hammer from Roots2 should be considered an Axe; that would probably be the easiest stop-gap solution. 

I don't know if it's possible, but perhaps even give it an add by wildcard search but only under the mod specification. That way, mods that add lots of variations on the same weapon can be covered with one config rather than 5-100 lines.

For example (not sure if this is the correct name, but you get the idea):

Instead of:
roots:wood_hammer
roots:stone_hammer
etc.

roots:*_hammer

**Comment by TheXFactor117** [comment]
This might be possible, though I'm not entirely sure. There would have to be a way to get an item from a ResourceLocation. I'll look to see if there is, and if so, I think I could actually implement it. Not too sure if the abbreviation would work though, that might not be possible.

I don't even know if the whole thing is even possible though at the moment.

**Comment by TheXFactor117** [comment]
It might be possible to do this actually. Only issue will be the larger the array in the config, the slower it will be. I'd like to have some way of not making it slow down if I wanted to implement such a system.

**Comment by Dragnier** [comment]
Perhaps a List instead of an Array? http://stackoverflow.com/questions/2391553/why-is-it-preferred-to-use-lists-instead-of-arrays-in-java

**Comment by TheXFactor117** [comment]
That won't solve anything really. Array's would be just as efficient. I'll have to think about it some more and see what the most efficient option is.
